main.go

```diff
@@ -0,0 +1,239 @@
+}
+
+func getMessages(c *gin.Context) {
+	c.IndentedJSON(http.StatusOK, messages)
+}
+
+func helperGetMessageById(id string) (*message, error) {
+	for i, m := range messages {
+		if m.ID == id {
+			return &messages[i], nil
+		}
+	}
+
+	return nil, errors.New("message not found")
+}
+
+func getMessageById(c *gin.Context) {
+	id := c.Param("id")
+	message, err := helperGetMessageById(id)
+
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "Message not found!"})
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, message)
+}
+
+func createMessage(c *gin.Context) {
+	var newMsg message
+
+	if err := c.BindJSON(&newMsg); err != nil {
+		return
+	}
+
+	newMsg.UUID = uuid.NewString()
+	newMsg.Created = time.Now().Format(time.RFC3339Nano)
+	newMsg.Type = "message"
+	newMsg.Upvote = 0
+	newMsg.Downvote = 0
+	newMsg.CommentCount = 0
+	newMsg.Deleted = false
+
+	messages = append(messages, newMsg)
+	c.IndentedJSON(http.StatusCreated, newMsg)
+}
+
+func addMessageUpvote(c *gin.Context) {
+	id, ok := c.GetQuery("id")
+
+	if !ok {
+		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "Missing id query parameter"})
+	}
+
+	message, err := helperGetMessageById(id)
+
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "Message not found!"})
+		return
+	}
+
+	message.Upvote += 1
+	c.IndentedJSON(http.StatusOK, message)
+}
+
+func removeMessageUpvote(c *gin.Context) {
+	id, ok := c.GetQuery("id")
+
+	if !ok {
+		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "Missing id query parameter"})
+	}
+
+	message, err := helperGetMessageById(id)
+
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "Message not found!"})
+		return
+	}
+
+	if message.Downvote <= 0 {
+		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "No upvotes to remove!"})
+		return
+	}
+
+	message.Downvote -= 1
+	c.IndentedJSON(http.StatusOK, message)
+}
+
+func addMessageDownvote(c *gin.Context) {
+	id, ok := c.GetQuery("id")
+
+	if !ok {
+		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "Missing id query parameter"})
+	}
+
+	message, err := helperGetMessageById(id)
+
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "Message not found!"})
+		return
+	}
+
+	message.Downvote += 1
+	c.IndentedJSON(http.StatusOK, message)
+}
+
+func removeMessageDownvote(c *gin.Context) {
+	id, ok := c.GetQuery("id")
+
+	if !ok {
+		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "Missing id query parameter"})
+	}
+
+	message, err := helperGetMessageById(id)
+
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "Message not found!"})
+		return
+	}
+
+	if message.Downvote <= 0 {
+		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "No downvotes to remove!"})
+		return
+	}
+
+	message.Downvote -= 1
+	c.IndentedJSON(http.StatusOK, message)
+}
+
+func deleteMessage(c *gin.Context) {
```

Comment:
This is a soft delete. Just toggles a boolean associate with the Content. If Delete: true for a content, the FE will know now to show it.